timemachine/fe/single_topology.py

```diff
@@ -356,49 +357,25 @@ def setup_end_state(ff, mol_a, mol_b, core, a_to_c, b_to_c):
         )
         # append idxs
         all_dummy_bond_idxs.extend(all_idxs[0])
-        all_dummy_angle_idxs.extend(all_idxs[1])
-        all_dummy_improper_idxs.extend(all_idxs[2])
         all_dummy_chiral_atom_idxs.extend(all_idxs[3])
         # append params
         all_dummy_bond_params.extend(all_params[0])
-        all_dummy_angle_params.extend(all_params[1])
-        all_dummy_improper_params.extend(all_params[2])
         all_dummy_chiral_atom_params.extend(all_params[3])
 
     # generate parameters for mol_a
     mol_a_top = topology.BaseTopology(mol_a, ff)
     mol_a_bond_params, mol_a_hb = mol_a_top.parameterize_harmonic_bond(ff.hb_handle.params)
```

Comment:
All that's needed from the constructed harmonic bond potentials are the bond idxs, I think (not the assigned Bond types).

A possible performance improvement, if needed, may be to get these idxs from `mol.GetBonds()` (wrapped for convenience in [`get_romol_bonds`](https://github.com/proteneer/timemachine/blob/1791077310d96a529653dc59941bab3c29939c59/timemachine/fe/utils.py#L385-L392)), rather than via [~90 specific bond smirks queries](https://github.com/proteneer/timemachine/blob/1791077310d96a529653dc59941bab3c29939c59/timemachine/ff/smirnoff_2.0.0.xml#L11-L98)).

Alternately, could make a version of the [`_sc.py` force field file](https://github.com/proteneer/timemachine/blob/1791077310d96a529653dc59941bab3c29939c59/timemachine/ff/params/smirnoff_1_1_0_sc.py#L41-L128) mentioned in [PR description](https://github.com/proteneer/timemachine/pull/1315#issue-2324253348) that only has a **single generic bond type** like `"[*:1]~[*:2]"`, so that the chiral consistency check still goes through the ff-typing code path `parameterize_harmonic_bond`, but just does fewer (unnecessary) smirks queries.

(Second option seems simpler and less error-prone.)